### PR TITLE
Feature/mst cli

### PR DESCRIPTION
**Pull request by grimadas**
## What is this pull request?
Command line instruments for multi-signatures transactions.
   
## Why do you implement it? Why do we need this pull request?
Need to sign uncompleted/multi-signature transactions 
  
## Details/Features
List of features / major commits
- Add 
- Refactor interactive cli
- Add interactive multisig 
- Add new query: get_uncompleted_transactions

### CODE REVIEW

File: iroha-cli/client.cpp [hunk 1]
```diff
@@ -17,16 +17,23 @@
...
-#include <utility>
-#include "model/converters/json_common.hpp"
 #include "model/converters/json_transaction_factory.hpp"
 #include "model/converters/pb_query_factory.hpp"
 #include "model/converters/pb_transaction_factory.hpp"
 
 namespace iroha_cli {
 
+  CliClient::CliClient(const CliClient &client)
+      : server_ip(client.server_ip),
```
Comment (l4l) [inline iroha-cli/client.cpp]: use [delegating constructor](http://en.cppreference.com/w/cpp/language/initializer_list#Delegating_constructor) instead
`: CliClient(client.server_ip, client.server_port) {}`

File: irohad/model/converters/impl/pb_query_response_factory.cpp [hunk 2]
```diff
@@ -216,7 +217,8 @@ namespace iroha {
       model::RolesResponse PbQueryResponseFactory::deserializeRolesResponse(
           const protocol::RolesResponse &response) const {
         model::RolesResponse res{};
-        std::copy(response.roles().begin(), response.roles().end(),
+        std::copy(response.roles().begin(),
+                  response.roles().end(),
                   res.roles.begin());
```
Comment (l4l) [inline irohad/model/converters/impl/pb_query_response_factory.cpp]: Can't find `res.roles` allocation, missed `std::back_inserter`?

File: iroha-cli/client.hpp [hunk 3]
```diff
@@ -59,6 +62,9 @@ namespace iroha_cli {
         std::string tx_hash);
 
    private:
+    std::string server_ip;
+    int server_port;
```
Comment (l4l) [inline iroha-cli/client.hpp]: 1. Both field seems only setted but unused
2. Can be const now (don't really sure that's it good semantically)

File: iroha-cli/interactive/impl/interactive_multisig.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      MenuPoints tx_menu;
+      iroha::model::converters::PbTransactionFactory pb_factory;
+      for (const auto &tx : txs.value().transactions()) {
+        auto iroha_tx = *pb_factory.deserialize(tx);
+        uncompleted_transactions.push_back(iroha_tx);
+        auto tx_hash = iroha::hash(tx);
+        auto decr = "creator " + iroha_tx.creator_account_id + " hash "
+            + tx_hash.to_hexstring() + " sigs/quorum "
+            + std::to_string(iroha_tx.signatures.size()) + "/"
+            + std::to_string(iroha_tx.quorum);
```
Comment (Warchant) [inline iroha-cli/interactive/impl/interactive_multisig.cpp]: never use concatenation with `to_string`. Always use `stringstream` instead:
```
std::stringstream ss;
ss << "creator " << tx_hash.to_hexstring() << iroha_tx.signatures.size(); // etc
```

File: iroha-cli/main.cpp [hunk 5]
```diff
@@ -99,79 +121,72 @@ int main(int argc, char *argv[]) {
...
+            FLAGS_name);
+        return EXIT_FAILURE;
+      }
+      // TODO 13/09/17 grimadas: Init counters from Iroha, or read from disk?
+      // IR-334
+      InteractiveCli interactiveCli(
+          FLAGS_name,
+          0,
+          0,
+          client,
```
Comment (Warchant) [inline iroha-cli/main.cpp]: add comments, which explain these magic values:
```
FLAGS_name,
/* something */ 0,
/* somethingelse */ 0,
```

File: iroha-cli/client.hpp (hunk 3 above)
Comment (grimadas) [reply iroha-cli/client.hpp]: They are used in copy-constructors and interactive cli to send to server.

File: irohad/model/converters/impl/pb_query_response_factory.cpp [hunk 6]
```diff
@@ -216,8 +217,9 @@ namespace iroha {
       model::RolesResponse PbQueryResponseFactory::deserializeRolesResponse(
           const protocol::RolesResponse &response) const {
         model::RolesResponse res{};
-        std::copy(response.roles().begin(), response.roles().end(),
-                  res.roles.begin());
+        std::copy(response.roles().begin(),
+                  response.roles().end(),
+                  std::back_inserter(res.roles));
```
Comment (Warchant) [inline irohad/model/converters/impl/pb_query_response_factory.cpp]: is it possible to write 
```
model::RolesResponse res = res.roles;
```
?

File: irohad/model/converters/impl/pb_query_response_factory.cpp [hunk 7]
```diff
@@ -235,7 +237,8 @@ namespace iroha {
       PbQueryResponseFactory::deserializeRolePermissionsResponse(
           const protocol::RolePermissionsResponse &response) const {
         model::RolePermissionsResponse res;
-        std::copy(response.permissions().begin(), response.permissions().end(),
+        std::copy(response.permissions().begin(),
+                  response.permissions().end(),
                   res.role_permissions.begin());
```
Comment (Warchant) [inline irohad/model/converters/impl/pb_query_response_factory.cpp]:
```
std::back_inserter
```
?
as in 
https://github.com/hyperledger/iroha/pull/706/files#diff-54506143c6fcf71e8b5954f1e82bf10fR222

File: iroha-cli/main.cpp [hunk 8]
```diff
@@ -99,79 +121,72 @@ int main(int argc, char *argv[]) {
...
+    else if (not FLAGS_json_query.empty()) {
       logger->info("Send query to {}:{}", FLAGS_address, FLAGS_torii_port);
       std::ifstream file(FLAGS_json_query);
       std::string str((std::istreambuf_iterator<char>(file)),
                       std::istreambuf_iterator<char>());
       iroha::model::converters::JsonQueryFactory serializer;
-      auto query_opt = serializer.deserialize(std::move(str));
+      auto query_opt = serializer.deserialize(str);
       if (not query_opt.has_value()) {
+        // TODO: Add specific reason why json is wrong
```
Comment (kamilsa) [inline iroha-cli/main.cpp]: mark who and when made this todo

**Pull request merged**